### PR TITLE
nixos/xmonad: adjust to reflect v0.17.0 update of xmonad

### PR DESCRIPTION
**Pull request by xaverdh**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sternenseemann — COMMENTED** [review]
Otherwise LGTM

File: nixos/modules/services/x11/window-managers/xmonad.nix [hunk 1]
```diff
@@ -128,33 +128,34 @@ in {
             [ ( (mod4Mask,xK_r), compileRestart True)
             , ( (mod4Mask,xK_q), restart "xmonad" True ) ]
 
+          compileRestart resume =
+            dirs <- io getDirectories
```
[the remark this answers is not in the document]
Comment (xaverdh) [reply nixos/modules/services/x11/window-managers/xmonad.nix:132]: I think `getDirectories` is of type `IO Directories` [here](https://hackage.haskell.org/package/xmonad-0.17.0/docs/XMonad-Core.html#v:getDirectories)?
This is where the directories are determined, you are probably mixing this up with the code that extracts them from `xmonad` internal state.

File: nixos/doc/manual/release-notes/rl-2205.section.md [hunk 2]
```diff
@@ -163,6 +163,10 @@ In addition to numerous new and upgraded packages, this release has the followin
   `useLLVM`. So instead of `(ghc.withPackages (p: [])).override { withLLVM = true; }`,
   one needs to use `(ghc.withPackages.override { useLLVM = true; }) (p: [])`.
 
+- The update of the haskell package set brings with it a new version of the `xmonad`
+  module, which will break your configuration if you use `lauch` as entrypoint. The
```
[the remark this answers is not in the document]
Comment (xaverdh) [reply nixos/doc/manual/release-notes/rl-2205.section.md]: thanks!

File: nixos/modules/services/x11/window-managers/xmonad.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (xaverdh) [reply nixos/modules/services/x11/window-managers/xmonad.nix:132]: Ah I see what you mean. Yes this will do the lookup logic twice.. will change that

**Comment by sternenseemann** [comment]
Thanks!

**Review by ivanbrennan — COMMENTED** [review]
Thanks!

**Pull request merged**